In [1]:
import pandas as pd
import numpy as np
from scipy.stats import chi2_contingency
import seaborn as sns

In [2]:
df = pd.read_csv('\\Users\\piyus\\Documents\\data assiogh\\Hypothesis_Testing_BuyerRatio-main\\BuyerRatio.csv')

In [3]:
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [4]:
df = df.drop(['Observed Values'], axis =1)

In [5]:
df

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [6]:
stat, p, dof, expected = chi2_contingency(df)

In [7]:
expected

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [8]:
p

0.6603094907091882

#Since the Pvalue (0.66) > alpha(0.050), we failed to reject Ho Hence the buying ratio of Male and Female are equal accross the regions
